src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/dashboardnotifications/defendant/CaseProceedOfflineDefendantNotificationHandler.java

```diff
@@ -67,4 +71,30 @@ public boolean shouldRecordScenarioInCaseProgression(CaseData caseData, boolean
             && caseMovedInCaseManStatesCaseProgression.contains(caseData.getPreviousCCDState())
             && isLipvLipOrLRvLip;
     }
+
+    @Override
+    protected void beforeRecordScenario(CaseData caseData, String authToken) {
+        dashboardApiClient.deleteNotificationsForCaseIdentifierAndRole(
+            caseData.getCcdCaseReference().toString(),
+            "DEFENDANT",
+            authToken
+        );
+
+        if (caseData.getGeneralApplications() != null && !caseData.getGeneralApplications().isEmpty()) {
+            caseData.getGeneralApplications()
```

Comment:
Do we need to do it for all general application types or only for Cosc application?